lib/pavilion/pav_test.py

```diff
@@ -178,6 +178,19 @@ def __init__(self, pav_cfg, config, sys_vars, _id=None):
         if _id is None:
             self.status.set(STATES.CREATED, "Test directory setup complete.")
 
+        # Checking validity of timeout values.
+        for loc in ['build', 'run']:
+            if loc in config and 'timeout' in config[loc] and \
+                    config[loc]['timeout'] is not None:
+                try:
+                    test_timeout = int(config[loc]['timeout'])
+                    if test_timeout < 0:
+                        raise ValueError()
+                except ValueError:
+                    raise PavTestError("{} timeout must be a non-negative "
+                            "integer or empty.  Received {}.".format(
+                                loc, config[loc]['timeout']))
```

Comment:
Completely normalize and convert the timeout here, and assign them to _build_timeout and _run_timeout. Then just use those later.